ci/check_diff.sh

```diff
@@ -89,11 +90,18 @@ function check_diff() {
         --unified=0 --no-index rustfmt_diff.txt feature_diff.txt 2>&1 | tail -n +6 | cut -c 2-
     )
 
+    # COPY diffs into output dir
+    mkdir $OUTPUT_DIR/$1
+    echo "Copying diffs to $OUTPUT_DIR/$1"
+    cp rustfmt_diff.txt $OUTPUT_DIR/$1/rustfmt_diff.txt
+    cp feature_diff.txt $OUTPUT_DIR/$1/feature_diff.txt
+
     if [ -z "$diff" ]; then
         echo "no diff detected between rustfmt and the feture branch"
         return 0
     else
-        echo "$diff"
+        echo "Copying diffs between rustfmt and feature branch to $OUTPUT_DIR/$1/diff.txt"
+        echo "$diff" >> $OUTPUT_DIR/$1/diff.txt
```

Comment:
> Not yet had a chance to really go through, but just wanted to check to confirm that the changes here are strictly additive and won't impact our ability to review the results without downloading and extracting an archive file?

@calebcartwright this change would result in the script no longer echoing the exact diff out. If you'd like I can add that back so you can still inspect the diff in the console. All other changes are additive.

Reply:
Thanks for the additional info. I want to make sure we don't make it more difficult (see: add too many extra steps) to use this in a standard PR context, so let's make sure we balance them